### PR TITLE
emacsPackages.yes-no: init at 2017-10-01

### PR DESCRIPTION
**Pull request by jcs090218**
###### Description of changes

Add emacs package, yes-no.el.

**Package description:** `Specify use of 'y-or-n-p' or 'yes-or-no-p' on a case-by-case basis`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation {
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix:3]: Use `trivialPackage` to package Emacs code.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages.nix [hunk 2]
```diff
@@ -232,6 +232,8 @@
   tree-sitter-langs = callPackage ./tree-sitter-langs { final = self; };
   tsc = callPackage ./tsc { };
 
+  yes-no = callPackage ./yes-no { };
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages.nix:235]: EmacsWiki codes are basically unmaintained. Why should we include this?

Comment (jcs090218) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages.nix:235]: I'm still using this package, and it doesn't break in any newer version of Emacs (I'm using snapshot `29.0.50`).

From your concern, I have set myself as the maintainer. I would keep an eye on this package if it does somehow break in the future. On the other hand, we can always place the fixed (update to date) version somewhere then replace the current `emacsmirror` version.

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix (hunk 1 above)
Comment (jcs090218) [reply pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix]: You mean `trivialBuild`?

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0";
+
+  src = fetchurl {
+    url = "https://github.com/emacsmirror/emacswiki.org/blob/185fdc34fb1e02b43759ad933d3ee5646b0e78f8/yes-no.el";
+    sha256 = "1k0nn619i82jiqm48k5nk6b8cv2rggh0i5075nhc85a2s9pwhx32";
+  };
+
+  dontUnpack = true;
+
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix:14]: Isn't this already done when using `trivialBuild`?

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib, fetchurl, trivialBuild }:
+
+trivialBuild {
+  pname = "yes-no";
+  version = "0";
+
+  src = fetchurl {
+    url = "https://github.com/emacsmirror/emacswiki.org/blob/185fdc34fb1e02b43759ad933d3ee5646b0e78f8/yes-no.el";
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix:8]:
```suggestion
    url = "https://raw.githubusercontent.com/emacsmirror/emacswiki.org/143bcaeb679a8fa8a548e92a5a9d5c2baff50d9c/yes-no.el";
```

This is the most recent version of this file.

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{ lib, fetchurl, trivialBuild }:
+
+trivialBuild {
+  pname = "yes-no";
+  version = "0";
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix:5]:
```suggestion
  version = "2017-10-01";
```

Given that this is a spare file, the version number into it makes few to no sense. Indeed it was modified two times and the Version line wasn't updated.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix (hunk 5 above)
Comment (jcs090218) [reply pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix]: Updated! Thanks!

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix (hunk 4 above)
Comment (jcs090218) [reply pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix]: Updated! Thanks!

File: pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix (hunk 3 above)
Comment (jcs090218) [reply pkgs/applications/editors/emacs/elisp-packages/yes-no/default.nix]: Ah... yes, you are right. Thanks!

**Pull request merged**